src/nunit.analyzers/TestMethodUsage/TestMethodUsageAnalyzer.cs

```diff
@@ -104,10 +113,41 @@ private static void AnalyzeAttribute(SyntaxNodeAnalysisContext context)
                         context.CancellationToken.ThrowIfCancellationRequested();
                         TestMethodUsageAnalyzer.AnalyzeExpectedResult(context, attributeNode, methodSymbol);
                     }
+
+                    var testMethodParameters = methodNode.ParameterList.Parameters.Count;
+                    var hasISimpleTestBuilderAttribute = HasISimpleTestBuilderAttribute(context.SemanticModel, methodNode.AttributeLists);
+                    var hasITestBuilderAttribute = HasITestBuilderAttribute(context.SemanticModel, methodNode.AttributeLists);
+                    var parametersMarkedWithIParameterDataSourceAttribute = ParametersMarkedWithIParameterDataSourceAttribute(
+                        context.SemanticModel, methodNode);
+
+                    if (testMethodParameters > 0 &&
+                        hasISimpleTestBuilderAttribute &&
+                        !hasITestBuilderAttribute &&
+                        parametersMarkedWithIParameterDataSourceAttribute < testMethodParameters)
+                    {
+                        context.ReportDiagnostic(
+                            Diagnostic.Create(
+                                simpleTestHasParameters,
+                                attributeNode.GetLocation(),
+                                testMethodParameters,
+                                parametersMarkedWithIParameterDataSourceAttribute));
+                    }
                 }
             }
         }
 
+        private static SyntaxNode? GetMethod(SyntaxNode node)
+        {
+            foreach (var ancestor in node.Ancestors())
+            {
+                if (ancestor.IsKind(SyntaxKind.ParameterList))
```

Comment:
Could you leave comment why we stop iteration if parameter list found?

Reply:
Actually, I'm not proud of this method, and I'm very open for improvements/suggestions. It is an attempt to only examine the attributes on the method and not the attributes on the parameters of the method.

The problem was that before this method we just took `context.Node.Ancestors().OfType<MethodDeclarationSyntax>().FirstOrDefault()` to find the method related to the attribute, but this both work for attributes on the method and attributes on the parameters of the method.
So for the following code we would mark both `[Test]` and `[Values(1, 2, 3)]` with the diagnostic even though I only intended to mark `[Test]`:
```csharp
[Test]
public void M(int p, [Values(1, 2, 3)] int q, int r)
{
    Assert.That(p, Is.EqualTo(q + r));
}
```

I'll also think of how to rephrase the code/this method.

Reply:
To begin with I've decided to replace the method - and the previous `context.Node.Ancestors().OfType<MethodDeclarationSyntax>` -  with `var declarationNode = context.Node.Parent?.Parent;` as this solves the problem and is less cryptic than the `GetMethod` method.